### PR TITLE
build: Update to Go 1.20, Alpine 3.17 and linter v1.51.2

### PR DESCRIPTION
**Pull request by lenny-intel**
**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **N/A**
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) **N/A**
  <link to docs PR>

## Testing Instructions
run `make docker`
run non-secure edgex stack with ds-virtual with dev images
verify services bootstrap and data is flowing to app-rules-engine
run secure edgex stack with ds-virtual with dev images
verify services bootstrap and data is flowing to app-rules-engine

### CODE REVIEW

File: cmd/security-spiffe-token-provider/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #  ----------------------------------------------------------------------------------
 
 # Build utility container
-ARG BUILDER_BASE=golang:1.18-alpine3.16
+ARG BUILDER_BASE=golang:1.20-alpine3.17
 FROM ${BUILDER_BASE} AS builder
 
 WORKDIR /edgex-go
```
Comment (jim-wang-intel) [inline cmd/security-spiffe-token-provider/Dockerfile:22]: can token provider also upgrade to alpine 3.17?

Comment (lenny-intel) [reply cmd/security-spiffe-token-provider/Dockerfile:22]: Yes, good catch.

Comment (lenny-intel) [reply cmd/security-spiffe-token-provider/Dockerfile:22]: fix, for this and a few others I missed.

**Review by jim-wang-intel — DISMISSED** [review]
LGTM

**Review by bnevis-i — CHANGES_REQUESTED** [review]
Please add

```
go version | sed -n -e 's/.*go\([0-9]\+\.[0-9]\+\.[0-9]\+\).*/\1/p' > .go-version && \
```

prior to the `make spire-whatever` line in the spire dockerfiles.

**Comment by lenny-intel** [comment]
> Please add
> 
> ```
> go version | sed -n -e 's/.*go\([0-9]\+\.[0-9]\+\.[0-9]\+\).*/\1/p' > .go-version && \
> ```
> 
> prior to the `make spire-whatever` line in the spire dockerfiles.

@bnevis-i . Done

**Review by bnevis-i — APPROVED** [review]
LGTM

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by ernestojeda — DISMISSED** [review]
LGTM

File: Makefile [hunk 2]
```diff
@@ -151,9 +151,11 @@ hadolint:
 	if which hadolint > /dev/null ; then hadolint --config .hadolint.yml `find * -type f -name 'Dockerfile*' -print` ; elif test "${ARCH}" = "x86_64" && which docker > /dev/null ; then docker run --rm -v `pwd`:/host:ro,z --entrypoint /bin/hadolint hadolint/hadolint:latest --config /host/.hadolint.yml `find * -type f -name 'Dockerfile*' | xargs -i echo '/host/{}'` ; fi
 	
 lint:
-	@which golangci-lint >/dev/null || echo "WARNING: go linter not installed. To install, run\n  curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b \$$(go env GOPATH)/bin v1.46.2"
+	@which golangci-lint >/dev/null || echo "WARNING: go linter not installed. To install, run make install-lint"
```
Comment (ernestojeda) [inline Makefile:154]: This is a nice way to handle missing dependencies.

**Comment by lenny-intel** [comment]
recheck

**Review by bnevis-i — APPROVED** [review]

**Review by bnevis-i — APPROVED** [review]

**Review by jim-wang-intel — APPROVED** [review]

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**